### ISSUE TITLE
Supersede `where()` in favour of anonymous functions

### ISSUE THREAD

**Issue by lionel-** [r-lib/tidyselect#248]
Anomymous functions are never ambiguous with variable names and so don't require `where()` to disambiguate. These would be equivalent:

```r
data %>% select(\(x) is.numeric(x))
data %>% select(~ is.numeric(.))
data %>% select(where(is.numeric))
```

**Comment by markfairbanks** [comment]
Would it be possible to make `select(~ is.numeric)` equivalent to `select(~ is.numeric(.))`?

**Comment by lionel-** [comment]
I don't think we should for the sake of consistency. We couldn't make it work e.g. in `map()` since `~ is.numeric` is a valid function that returns a function.

**Comment by SimonDedman** [comment]
So to be crystal clear, this would be replacing `select(where(is.numeric))` with `select(~ is.numeric(.))` as the most common/intuitive/well known usage case?

**Comment by lionel-** [comment]
Until we allow anonymous functions, you can use:

```r
dplyr::select(dplyr::across(where(~ is.numeric(.)))
```

And then somewhere in your package, e.g. in `utils.R` or `mypkg-package.R`:

```r
utils::globalVariables("where")
```

**Comment by SimonDedman** [comment]
Good stuff, thank you.

**Comment by lionel-** [comment]
Since it's been so long we added `where()` and it's now used in a lot of existing code, we've decided to keep it as the official syntax for predicate functions. We'll export it from tidyselect and dplyr soon.

**Comment by yutannihilation** [comment]
Glad to see `where()` is exported at last!

Do you think #202 is worth reopening? I think closing this means we can explore new additions to the current `where()` syntax.
